src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -377,19 +378,16 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         this.membershipId = undefined;
         this.emit(MatrixRTCSessionEvent.JoinStateChanged, false);
 
-        const timeoutPromise = new Promise((r) => {
-            if (timeout) {
-                // will never resolve if timeout is not set
-                setTimeout(r, timeout, "timeout");
-            }
-        });
-        return new Promise((resolve) => {
-            Promise.race([this.triggerCallMembershipEventUpdate(), timeoutPromise]).then((value) => {
-                // The timeoutPromise returns the string 'timeout' and the membership update void
-                // A success implies that the membership update was quicker then the timeout.
-                resolve(value != "timeout");
-            });
-        });
+        if (timeout) {
+            // The sleep promise returns the string 'timeout' and the membership update void
+            // A success implies that the membership update was quicker then the timeout.
+            return (
+                (await Promise.race([this.triggerCallMembershipEventUpdate(), sleep(timeout, "timeout")])) !== "timeout"
+            );
+        } else {
+            await this.triggerCallMembershipEventUpdate();
+            return false;
```

Comment:
I think this should be `true`?

It doesn't seem to be covered by unit tests...

Reply:
Oops, yes!  Fixed & somewhat tested in 0383056de